`TestIntegrations/Discovery` flakiness

## Failure

#### Link(s) to logs

- https://github.com/gravitational/teleport/actions/runs/5043488987/jobs/9045336040#step:6:2379
- v12 https://github.com/gravitational/teleport/actions/runs/5623334769/job/15237803399
- https://github.com/gravitational/teleport/actions/runs/7009114863/job/19066880309?pr=34961

#### Relevant snippet

```
    integration_test.go:3900: 
        	Error Trace:	/__w/teleport/teleport/integration/integration_test.go:3900
        	            				/__w/teleport/teleport/integration/integration_test.go:114
        	Error:      	Received unexpected error:
        	            	failed connecting to host 127.0.0.1:39505: failed to dial target host
        	            		unable to fetch access point, this proxy clusterPeer(TunnelConnection(name=2f33b91b-5b99-43e7-b35e-37c32bcfec48-cluster-remote, type=, cluster=cluster-remote, proxy=2f33b91b-5b99-43e7-b35e-37c32bcfec48)) has not been discovered yet, try again later
        	Test:       	TestIntegrations/Discovery
```

```
     integration_test.go:3664: 
        	Error Trace:	/__w/teleport/teleport/integration/integration_test.go:3664
        	            				/__w/teleport/teleport/integration/integration_test.go:109
        	Error:      	Received unexpected error:
        	            	timed out, only 0/2 events received. received: [], expected: [ProxyReverseTunnelReady ProxySSHReady]
        	Test:       	TestIntegrations/Discovery
```

Another in the merge queue to master https://github.com/gravitational/teleport/actions/runs/7090182023/job/19296583245
```
=== RUN   TestIntegrations/Discovery
    integration_test.go:4172: 
        	Error Trace:	/__w/teleport/teleport/integration/integration_test.go:6155
        	            				/opt/go/src/runtime/asm_amd64.s:1650
        	Error:      	Expected value not to be nil.
    integration_test.go:4172: 
        	Error Trace:	/__w/teleport/teleport/integration/integration_test.go:6151
        	            				/__w/teleport/teleport/integration/integration_test.go:4172
        	            				/__w/teleport/teleport/integration/integration_test.go:126
        	Error:      	Condition never satisfied
        	Test:       	TestIntegrations/Discovery
    --- FAIL: TestIntegrations/Discovery (50.00s)
```

--- a/integration/helpers/trustedclusters.go
+++ b/integration/helpers/trustedclusters.go
@@ -28,6 +28,7 @@ import (
 	log "github.com/sirupsen/logrus"
 	"github.com/stretchr/testify/require"
 
+	"github.com/gravitational/teleport"
 	"github.com/gravitational/teleport/api/defaults"
 	"github.com/gravitational/teleport/api/types"
 	"github.com/gravitational/teleport/api/utils/retryutils"
@@ -148,7 +149,7 @@ func WaitForActiveTunnelConnections(t *testing.T, tunnel reversetunnelclient.Ser
 		if err != nil {
 			return false
 		}
-		return cluster.GetTunnelsCount() >= expectedCount
+		return cluster.GetTunnelsCount() >= expectedCount && cluster.GetStatus() == teleport.RemoteClusterStatusOnline
 	},
 		30*time.Second,
 		time.Second,

--- a/integration/integration_test.go
+++ b/integration/integration_test.go
@@ -4114,6 +4114,8 @@ func testDiscovery(t *testing.T, suite *integrationTestSuite) {
 	helpers.WaitForActiveTunnelConnections(t, main.Tunnel, "cluster-remote", 1)
 	helpers.WaitForActiveTunnelConnections(t, secondProxy, "cluster-remote", 1)
 
+	waitForNodesToRegister(t, main, "cluster-remote")
+
 	// execute the connection via first proxy
 	cfg := helpers.ClientConfig{
 		Login:   username,

